org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/converter/ZigBeeChannelConverterProvider.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.zigbee.converter;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeUID;
+
+/**
+ * The {@link ZigBeeChannelConverterProvider} can be registered as OSGi service in order to make additional
+ * {@link ZigBeeBaseChannelConverter}s available to the binding.
+ *
+ * @author Thomas Höfer - Initial contribution
+ */
+@NonNullByDefault
+public interface ZigBeeChannelConverterProvider {
+
+    /**
+     * Provides a map of {@link ChannelTypeUID}s and their corresponding channel converters.
+     *
+     * @return all channel converters of this provider
+     */
+    Map<ChannelTypeUID, Class<? extends ZigBeeBaseChannelConverter>> getChannelConverters();
+
+    /**
+     * Provides the map of all channels to be consolidated. The binding creates a hierarchical list of channels that
+     * are to be removed due to inheritance.
+     *
+     * It is to be noted that the order is important in the event that there are multiple removals, e.g. we want to
+     * remove switch before dimmer, then dimmer if color is present - if device creates both channels, then we keep the
+     * one on the right (i.e. map.value):
+     * {code}
+     * channelConsolidation.put(ZigBeeBindingConstants.CHANNEL_SWITCH_LEVEL,
+     * ZigBeeBindingConstants.CHANNEL_SWITCH_ONOFF);
+     * channelConsolidation.put(ZigBeeBindingConstants.CHANNEL_COLOR_COLOR,
+     * ZigBeeBindingConstants.CHANNEL_SWITCH_LEVEL);
+     * {code}
+     *
+     * @return the map of all channels to be consolidated
+     */
+    default Map<ChannelTypeUID, ChannelTypeUID> getChannelConsolidations() {
+        return new HashMap<ChannelTypeUID, ChannelTypeUID>();
```

Comment:
i don't really understand why this method is required? Surely this should be a "static" definition - ie the channel consolidation is a function of ESH channel type inheritance and I don't know why converters should ever need to add new consolidations?  This should only be a function of the binding and the framework and should not need to be something the converters worry about I think?

I also don't think it will work to have channel providers asynchronously adding consolidation rules as the order is important and weird things will happen if that is not respected.

Reply:
Currently I dont have any reason to provide additional consolidations. I thought it could make sense to provide additional consolidations how the converter factory is currently implemented.

I will remove the operation.